lib/AST/ASTWalker.cpp

```diff
@@ -1251,7 +1251,7 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
           D->getDeclContext()->getParentSourceFile())
         return true;
       if (Decl *ParentD = Walker.Parent.getAsDecl())
-        return (isa<NominalTypeDecl>(ParentD) || isa<ExtensionDecl>(ParentD));
+        return (!D->hasClangNode() && (isa<NominalTypeDecl>(ParentD) || isa<ExtensionDecl>(ParentD)));
```

Comment:
Did checking `D->getParentPatternBinding()` not work here? I'm concerned we'll still visit some VarDecls twice because the ClangImporter does seem to create PBDs for some VarDecls.

Reply:
Casting to a VarDecl and checking for `VD->getParentPatternBinding()` caused availability checking to start erroneously failing, preventing the standard library from building (and failing a bunch of tests as well).

Reply:
Should it only check for the parent pattern binding if the parent is a decl, i.e. in the branch i edited?